src/ucm/util/reloc.c

```diff
@@ -238,10 +252,17 @@ static int ucm_reloc_phdr_iterator(struct dl_phdr_info *info, size_t size, void
 /* called with lock held */
 static ucs_status_t ucm_reloc_apply_patch(ucm_reloc_patch_t *patch)
 {
-    ucm_reloc_dl_iter_context_t ctx = {
-        .patch  = patch,
-        .status = UCS_OK
-    };
+    ucm_reloc_dl_iter_context_t ctx;
+    int                         success;
+
+    /* Find default shared object, usually libc */
+    success = dladdr(malloc_trim, &ctx.def_dlinfo);
```

Comment:
why malloc_trim and not our function?

Reply:
just random function from libc, this is not the best choice because we have our replacement.. will fix with `getpid`.
we can't use our function because we have to call some original if something was not handled by ours hooks, for example.

Reply:
seems a bit hacky to use getpid().. is there some more elegant way? @hoopoepg @Artemy-Mellanox what do you think?